### PR TITLE
Allow rustic-compile to accept a command argument

### LINKED ISSUE #57: rustic-compile always prompts interactively (even when called from elisp)
### Experience behavior

The compile method of Rustic differs to the built-in one: it is always interactively prompting for a compile-command, no matter if it's called interactively or programmatically.

Example 1:

```lisp=
(rustic-compile "cargo build")
```

Here I provided my own compile command, and I expect the compilation to start immediately. But it doesn't, I'm getting needlessly asked for a compile-command. This makes (rustic-compile ...) unusable for scripting in Elisp.

Example 2:

```lisp=
(let ((compilation-read-command nil))
 (rustic-compile "cargo build"))
```

Even that generates a prompt for the interactive command.

### Expected Behavior

Contrast this with

```lisp=
(compile "cargo build")
```

which doesn't prompt at all and just does what it is called to do: compile my source. That is: the provided optional argument overrides the variable compile-command, and makes it not prompt. The "not prompting" btw also happens when compilation-read-command is t.

### Version of rustic-mode

This is on rustic-mode HEAD d765680373234a6c231acf20c76b07422afcfdf9

### PR DESCRIPTION
**Pull request by CeleritasCelery**
Fixes #57

### CODE REVIEW

**Comment by CeleritasCelery** [comment]
Looking at the tests from the PR, this function was not intended to be an analog to `compile`. The "right" way to call it from elisp would be to bind `compilation-arguments` and then call the function with no arguments. We could change it to align more with `compile` and take a command as an argument (which I think is a better design), but then we risk breaking code that is already using it the current way.

File: test/rustic-compile-test.el [hunk 1]
```diff
@@ -85,12 +85,25 @@
...
-    (should-not compilation-directory)
-    (should-not compilation-arguments)
-    (setq compilation-arguments "cargo fmt")
+    (let* ((default-directory dir)
+           (proc (rustic-compile "cargo fmt")))
+      (should (process-live-p proc))
+      (while (eq (process-status proc) 'run)
+        (sit-for 0.1))
+      (should (string= compilation-directory dir))
+      (let ((proc (rustic-recompile)))
```
Comment (psibi) [inline test/rustic-compile-test.el:94]: Why are we doing `rustic-recompile` here ? From a quick look the previous code doesn't seem to have it.

File: rustic-compile.el [hunk 2]
```diff
@@ -537,21 +537,19 @@ buffer."
...
+Interactively, prompts for the command if the variable
+`compilation-read-command' is non-nil; otherwise uses
+`rustic-compile-command'."
+  (interactive
+   (list
+    (if compilation-read-command
+        (compilation-read-command (or (car compilation-arguments)
+                                      (rustic-compile-command)))
+      (rustic-compile-command))))
+  (rustic-set-compilation-arguments command)
```
Comment (psibi) [inline rustic-compile.el:552]: The new behavior is nice!

File: test/rustic-compile-test.el (hunk 1 above)
Comment (CeleritasCelery) [reply test/rustic-compile-test.el:94]: The previous code does have it

https://github.com/emacs-rustic/rustic/blob/0d79e4c19b7b77960a7665b8cef181954e4023e5/test/rustic-compile-test.el#L98

Comment (CeleritasCelery) [reply test/rustic-compile-test.el:94]: I added a new test for compiling interactively rather than via lisp, so the diff looks a little weird.

**Review by psibi — APPROVED** [review]

**Pull request merged**